stardict.cc

```diff
@@ -466,20 +466,16 @@ string StardictDictionary::handleResource( char type, char const * resource, siz
     {
       QString articleText = QString( "<div class=\"sdct_h\">" ) + QString::fromUtf8( resource, size ) + "</div>";
 
-      const char * srcRegexpStr = "(<\\s*img\\s+[^>]*src\\s*=\\s*[\"']+)(?!(?:data|https?|ftp):)";
-      const char * hrefRegexpStr = "(<\\s*link\\s+[^>]*href\\s*=\\s*[\"']+)(?!(?:data|https?|ftp):)";
+      const char * srcRegexpStr = "(<\\s*(?:img|script)\\s[^>]*src\\s*=\\s*[\"']*)(?![^>]*(?:data|https?|ftp):)";
+      const char * hrefRegexpStr = "(<\\s*link\\s[^>]*href\\s*=\\s*[\"']*)(?![^>]*(?:data|https?|ftp):)";
```

Comment:
Why insert `[^>]*` in both regex strings?

Reply:
For proper negative look ahead (`data|https?|ftp`)  with optional quotes regex should have possessive modifier which is not supported by older `QRegExp`. Part `[^>]*` ensures proto goes before the closing `>`. 

Otherwise it will be false-positives for e.g. `<link href='http://example.com>` which "false-positives" mention is about.

Reply:
But inserting `[^>]*` is an extra risk, isn't it? If you don't care about Qt 4, maybe fix only the Qt 5 regex in a safer (and more efficient?) way?

Reply:
I see no reason for new false-positives with the `[^>]*`. It may be false-negatives for cases like `<link href='prefix_http:'>`, which should not be a problem.

Also does difference in Qt4 and Qt5 logic will be OK?

Reply:
Another possibility of false positive: `<img src="..." another_attribute="http:...">`

@Abs62, would you be OK with a safer support for omitting quotes (which is valid HTML) only in Qt 5? By safer, I mean the possessive modifier in the regex instead of the inserted `[^>]*`.

Reply:
@bergentroll, would the following changes perhaps work and be 100% safe?
1. Reorder two consecutive blocks of code: `\\s*[\"']*` and `)(?!`;
2. Remove the inserted `[^>]*`;
3. Capture the moved spaces and quotes and adjust `articleText.replace` code below accordingly.

Reply:
Awesome! Sādhu!

Your proposal [do the job fine](https://regex101.com/r/z7PanX/2). Already checked and pushed.